dpypelines/pipeline/dataset_ingress_v1.py

```diff
@@ -187,25 +187,29 @@ def dataset_ingress_v1(files_dir: str, pipeline_config: dict):
     for supp_dist_pattern in supp_dist_patterns:
         try:
             if not local_store.has_lone_file_matching(supp_dist_pattern):
-                err = FileNotFoundError(
-                    f"No file found matching pattern {supp_dist_pattern}"
-                )
-                email_content = supplementary_distribution_not_found_email(
-                    supp_dist_pattern
-                )
-                email_client.send(
-                    submitter_email, email_content.subject, email_content.message
-                )
-                logger.error(
-                    "Email sent to submitter about missing supplementary distribution file",
-                    err,
-                    data={
-                        "submitter_email": submitter_email,
-                        "supplementary_distribution_pattern": supp_dist_pattern,
-                    },
-                )
-                de_notifier.failure()
-                raise err
+                # Catch a trivial raise as we need the stack trace of the
+                # error for the logger, so it needs to be a raised error.
+                try:
+                    raise FileNotFoundError(
```

Comment:
Do we want to send an email at this point?

Reply:
we sent if just below.

this is the bit I wrong about before - we need to have a raised error (so the logger can create info from the stack trace).

so we:
- raise it
- catch it
- log it
- send email
- raise it properly.

you could email just before here like you say, then raise and catch it to do the logger part, but seemed slightly neater having just the raise.

did that make sense or am I misunderstanding you?

Reply:
Makes sense, I've just never seen the `try: raise, except: raise again` structure before.